mypy/typeanal.py

```diff
@@ -171,6 +175,11 @@ def __init__(self,
         self.is_typeshed_stub = is_typeshed_stub
         self.warn_bound_tvar = warn_bound_tvar
         self.third_pass = third_pass
+        # Names of type aliases encountered while analysing a type will be collected here.
+        # (This also recursively includes the names of aliases they depend on.)
+        # TODO: adding the sub-dependencies shouldn't be necessary since the logic
+        # of fine-grained should propagate triggered dependencies.
```

Comment:
Why is this necessary? It would be useful to understand the precise root cause for why this is included and mention it in the TODO comment.